src/coreclr/hosts/corerun/corerun.cpp

```diff
@@ -210,6 +211,35 @@ static void log_error_info(const char* line)
     std::fprintf(stderr, "%s\n", line);
 }
 
+size_t HOST_CONTRACT_CALLTYPE get_runtime_property(
+    const char* key,
+    char* value_buffer,
+    size_t value_buffer_size,
+    void* contract_context)
+{
+    configuration* config = static_cast<configuration *>(contract_context);
+
+    if (::strcmp(key, HOST_PROPERTY_ENTRY_ASSEMBLY_NAME) == 0)
+    {
+        // Compute the entry assembly name based on the entry assembly full path
+        pal::string_t dir;
+        pal::string_t file;
+        pal::split_path_to_dir_filename(config->entry_assembly_fullpath, dir, file);
+        file = file.substr(0, file.rfind(W('.')));
+
+        pal::string_utf8_t file_utf8 = pal::convert_to_utf8(file.c_str());
+        size_t len = file_utf8.size() + 1;
+        if (value_buffer_size < len)
+            return len + 1;
```

Comment:
The `+1` seems wrong here - we already added `+1` above to the length of the string. And also below we only require `len` length of the array.

Reply:
Oops, I definitely failed at updating after switching `len` from not having the `+1` to having it.